### PR TITLE
chore(TCOMP-2092): improve mvn coordinate resolver

### PR DESCRIPTION
**Pull request by ozhelezniak-talend**
* add case with classifier

### Requirements
https://jira.talendforge.org/browse/TCOMP-2092

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant

### CODE REVIEW

**Review by undx — DISMISSED** [review]
minor comments but thanks for this more robust converter ;-)

File: container/container-core/src/main/java/org/talend/sdk/component/dependencies/maven/MvnCoordinateToFileConverter.java [hunk 1]
```diff
@@ -43,18 +48,27 @@ public Artifact toArtifact(final String coordinates) {
         }
 
         switch (segments.length) { // support some optional values 3: g:a:v, 4: g:a:t:v
```
[the remark this answers is not in the document]
Comment (ozhelezniak-talend) [reply container/container-core/src/main/java/org/talend/sdk/component/dependencies/maven/MvnCoordinateToFileConverter.java:50]: of course, thank you. didn't see it(

File: container/container-core/src/main/java/org/talend/sdk/component/dependencies/maven/MvnCoordinateToFileConverter.java [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+ * <p>
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- *
+ * <p>
  * http://www.apache.org/licenses/LICENSE-2.0
- *
+ * <p>
  * Unless required by applicable law or agreed to in writing, software
```
[the remark this answers is not in the document]
Comment (ozhelezniak-talend) [reply container/container-core/src/main/java/org/talend/sdk/component/dependencies/maven/MvnCoordinateToFileConverter.java:10]: ok

[the remark this answers is not in the document]
Comment (ozhelezniak-talend) [reply container/container-core/src/main/java/org/talend/sdk/component/dependencies/maven/MvnCoordinateToFileConverter.java]: I used talend-eclipse formatter settings and it moves the switch on one tab to left.
Looks like the component-runtime uses talend-eclipse formatted too but the formatting is different (switch expressions)

**Review by undx — APPROVED** [review]
LGTM

**Pull request merged**